recipes/autochanger_common.rb

```diff
@@ -0,0 +1,29 @@
+# Add default repos for Bareos Packages
+include_recipe 'bareos::package_repos_common'
+
+# Manage mtx-changer config http://doc.bareos.org/master/html/bareos-manual-main-reference.html#QQ2-1-467
+package 'bareos-storage-tape'
+
+execute 'mtx-changer script' do
+  command '/usr/bin/bareos/scripts/mtx-changer'
+  creates '/etc/bareos/mtx-changer.conf'
+  action :run
+end
+
+template 'mtx-changer config' do
+  source 'mtx_changer.erb'
+  path '/etc/bareos/mtx-changer.conf'
+  owner 'root'
+  group 'root'
+  mode '0644'
+end
```

Comment:
why do you create this file twice?

Reply:
The first script does a few different things that are more easily managed via the script then I manage the one file via template to overwrite the one. The script step is required though for the most part without having to go through more pain..

Reply:
The creates bit just makes sure the resource is only executed in the case that the file doesn't exist.

Reply:
is a re-execution of mtx-changer script going to cause changes that the template is going to continuously have to fix?

Reply:
ah. ok.

Reply:
If it runs then yes i believe so. That is why I am preventing it from running unless the file exists.